thread.c

```diff
@@ -597,11 +665,12 @@ void dispatch_conn_new(int sfd, enum conn_states init_state, int event_flags,
         return;
     }
 
-    int tid = (last_thread + 1) % settings.num_threads;
-
-    LIBEVENT_THREAD *thread = threads + tid;
+    LIBEVENT_THREAD *thread;
 
-    last_thread = tid;
+    if (settings.num_napi_ids)
```

Comment:
I'd probably flip this so the default branch is the common case (round robin) but I'm not going to be strict about it here.

Reply:
ok.